puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -73,6 +75,41 @@ def build_events_list
     end
   end
 
+  # @return Array[Hash]
+  # @api private
+  def build_logs_list
+    profile("Build logs list (count: #{logs.count})",
+            [:puppetdb, :logs_list, :build]) do
+      log_list = []
+      logs.each do |log|
+        log_hash = {
+          'file' => log.file,
+          'line' => log.line,
+          'level' => log.level,
+          'message' => log.message,
+          'source' => log.source,
+          'tags' => [*log.tags],
+          'time' => Puppet::Util::Puppetdb.to_wire_time(log.time),
+        }
+        log_list.push(log_hash)
+      end
+
+      log_list
+    end
+  end
+  def build_metrics_list
+    profile("Build metrics list (count: #{metrics.count})",
```

Comment:
I believe metrics.count will always be 4, since its a fixed set of keys at the top level of the hash.